presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/KinesisTableDescriptionSupplier.java

```diff
@@ -0,0 +1,126 @@
+
+    private final KinesisConfig kinesisConfig;
+    private final JsonCodec<KinesisStreamDescription> streamDescriptionCodec;
+    private final S3TableConfigClient s3TableConfigClient;
+
+    @Inject
+    public KinesisTableDescriptionSupplier(KinesisConfig kinesisConfig,
+            JsonCodec<KinesisStreamDescription> streamDescriptionCodec,
+            S3TableConfigClient s3TableConfigClient)
+    {
+        this.kinesisConfig = requireNonNull(kinesisConfig, "kinesisConfig is null");
+        this.streamDescriptionCodec = requireNonNull(streamDescriptionCodec, "streamDescriptionCodec is null");
+        this.s3TableConfigClient = requireNonNull(s3TableConfigClient, "S3 table config client is null");
+    }
+
+    @Override
+    public Map<SchemaTableName, KinesisStreamDescription> get()
+    {
+        if (this.s3TableConfigClient.isUsingS3()) {
+            return this.s3TableConfigClient.getTablesFromS3();
+        }
+        else {
+            return getTablesFromDirectory();
+        }
+    }
+
+    public Map<SchemaTableName, KinesisStreamDescription> getTablesFromDirectory()
+    {
+        ImmutableMap.Builder<SchemaTableName, KinesisStreamDescription> builder = ImmutableMap.builder();
+        try {
+            for (Path file : listFiles(Paths.get(kinesisConfig.getTableDescriptionLocation()))) {
+                if (Files.isRegularFile(file) && file.getFileName().toString().endsWith("json")) {
+                    KinesisStreamDescription table = streamDescriptionCodec.fromJson(Files.readAllBytes(file));
+                    String schemaName = firstNonNull(table.getSchemaName(), kinesisConfig.getDefaultSchema());
+                    log.debug("Kinesis table %s %s %s", schemaName, table.getTableName(), table);
+                    builder.put(new SchemaTableName(schemaName, table.getTableName()), table);
+                }
+            }
+
+            Map<SchemaTableName, KinesisStreamDescription> tableDefinitions = builder.build();
+            log.debug("Loaded table definitions: %s", tableDefinitions.keySet());
+
+            return tableDefinitions;
+        }
+        catch (IOException e) {
+            log.warn(e, "Error: ");
+            throw Throwables.propagate(e);
+        }
+    }
+
+    /**
+     * Shutdown any periodic update jobs.
+     */
+    @PreDestroy
+    public void shutdown()
+    {
+        this.s3TableConfigClient.run();
+        return;
+    }
+
+    private static List<Path> listFiles(Path dir)
+    {
+        if ((dir != null) && Files.isDirectory(dir)) {
+            try (DirectoryStream<Path> stream = Files.newDirectoryStream(dir)) {
+                ImmutableList.Builder<Path> builder = ImmutableList.builder();
+                for (Path file : stream) {
+                    builder.add(file);
+                }
+
+                return builder.build();
+            }
+            catch (IOException | DirectoryIteratorException x) {
+                log.warn(x, "Warning.");
```

Comment:
The exception is being re-thrown, so no point in logging it.

Reply:
changed this